### PR TITLE
Split in geopandas-base and geopandas outputs

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
This creates two outputs: `geopandas-base` with the minimal dependencies, and `geopandas` which additionally depends on rtree, fiona and matplotlib-base (we previously did not include matplotlib in the dependencies, but with the split and the availability of matplotlib-base, I think we can add it back)

xref https://github.com/geopandas/geopandas/issues/1261, https://github.com/geopandas/geopandas/issues/1313

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@conda-forge-admin, please rerender

**Comment by jorisvandenbossche** [comment]
@conda-forge-admin, please rerender

**Comment by jorisvandenbossche** [comment]
@conda-forge-admin, please rerender

**Comment by jorisvandenbossche** [comment]
Some issues:

- It seems to have added builds for windows/linux/osx, so the `noarch` no longer works?
- The build error is `'PYTHON' is undefined`. Is that not supported in a nested build script (nested in `outputs:`)?

**Comment by jorisvandenbossche** [comment]
@conda-forge-admin, please add noarch: python

**Comment by martinfleis** [comment]
What about `mapclassify`? I'd vote for adding it alongside matplotlib, even though it comes with some other [dependencies](https://github.com/pysal/mapclassify/blob/master/requirements.txt#L1-L5).

With upcoming `explore`, I assume we will want to have it supported out of the box with `conda install geopandas` and that depends on mapclassify even for non-`scheme` choropleths, so we will be likely adding it soon anyway.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,36 +1,56 @@
...
-    - geopandas
-    - geopandas.io
-  files:
-    - test_data
+  number: 1
+  noarch: python
+
+outputs:
+  - name: geopandas-base
+    script: {{ PYTHON }} -m pip install . --no-deps -vv
```
Comment (ocefpaf) [inline recipe/meta.yaml:18]: Sadly conda-build `script` in multiple outputs does work like `script` in a single output. We must point it to a [script file like  this](https://github.com/conda-forge/importlib_metadata-feedstock/blob/master/recipe/meta.yaml).

**Comment by jorisvandenbossche** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (jorisvandenbossche) [reply recipe/meta.yaml]: Thanks for the pointer! That's working now

**Review by ocefpaf — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@martinfleis hmm, yes `mapclassify` would probably be good to add as well, but that indeed brings in scipy and scikit-learn ..

**Comment by martinfleis** [comment]
> but that indeed brings in scipy and scikit-learn

You can always go with `geopandas-base` if you don't want them. Most people won't mind and those who do have another option.

**Comment by jorisvandenbossche** [comment]
OK, let's see how this goes ;)

**Pull request merged**